corehq/apps/users/models.py

```diff
@@ -2356,6 +2358,25 @@ def create(cls, domain, username, password, created_by, created_via, email=None,
                                  by_domain_required_for_log=by_domain_required_for_log)
         return web_user
 
+    def add_domain_membership(self, domain, timezone=None, **kwargs):
+        if TABLEAU_USER_SYNCING.enabled(domain):
+            # Inline import avoids a circular import error
+            from corehq.apps.reports.util import add_tableau_user
+            from corehq.apps.reports.models import TableauUser
+            try:
+                add_tableau_user(domain, self.username)
+            except (TableauAPIError, TableauUser.DoesNotExist) as e:
+                notify_exception(None, str(e), details={
+                    'domain': domain
+                })
+        super().add_domain_membership(domain, timezone, **kwargs)
```

Comment:
I'm not a fan of overriding these methods for the sake of a non-core feature. An alternative would be to use signals to hook up with tableau syncing but that also has it's own drawbacks so I'm not suggesting a change, just noting my lack of delight.